tests/integration/kernel/src/main.c

```diff
@@ -0,0 +1,305 @@
+					thread_entry_fn_fifo, &fifo, NULL, NULL,
+					K_PRIO_PREEMPT(0), K_INHERIT_PERMS, K_NO_WAIT);
+
+	/* Let the thread B run */
+	k_sem_take(&sema, K_FOREVER);
+
+	/* Get items from fifo */
+	for (i = 0U; i < LIST_LEN; i++) {
+		rx_data = k_fifo_get(&fifo, K_NO_WAIT);
+		zassert_equal(rx_data->value, fifo_data[i].value);
+	}
+
+	/* Clear the spawn thread */
+	k_thread_abort(tid);
+}
+
+/* LIFO SCENARIO */
+/* Thread A enters items into a lifo, starts Thread B and waits for a semaphore. */
+/* Thread B extracts all items from the lifo and enters some items back into the lifo. */
+/* Thread B gives the semaphore for Thread A to continue. */
+/* Once the control is returned back to Thread A, it extracts all items from the liifo. */
+/* Verify the data's correctness. */
+
+struct lifo_item_t {
+	void *LIFO_reserved;   /* 1st word reserved for use by LIFO */
+	uint8_t value;
+};
+static struct lifo_item_t lifo_data[LIST_LEN];
+K_LIFO_DEFINE(lifo);
+
+
+static void thread_entry_fn_lifo(void *p1, void *p2, void *p3)
+{
+	struct lifo_item_t  *rx_data;
+	uint32_t i;
+
+	/* Get items from lifo */
+	for (i = 0U; i < LIST_LEN; i++) {
+		rx_data = k_lifo_get((struct k_lifo *)p1, K_NO_WAIT);
+		zassert_equal(rx_data->value, lifo_data[LIST_LEN-1-i].value);
+	}
+
+	/* Put items into lifo */
+	for (i = 0U; i < LIST_LEN; i++) {
+		lifo_data[i].value *= i;
+		k_lifo_put((struct k_lifo *)p1, &lifo_data[i]);
+	}
+
+	/* Give control back to Thread A */
+	k_sem_give(&sema);
+}
+
+ZTEST(kernel, test_lifo_usage)
+{
+	struct lifo_item_t *rx_data;
+	uint32_t i;
+
+	/* Init binary semaphore */
+	k_sem_init(&sema, 0, 1);
+
+	/* Set and Put items into lifo */
+	for (i = 0U; i < LIST_LEN; i++) {
+		lifo_data[i].value = i;
+		k_lifo_put(&lifo, &lifo_data[i]);
+	}
+
+	/* Create the Thread B */
+	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
+					thread_entry_fn_lifo, &lifo, NULL, NULL,
+					K_PRIO_PREEMPT(0), K_INHERIT_PERMS, K_NO_WAIT);
+
+	/* Let the thread B run */
+	k_sem_take(&sema, K_FOREVER);
+
+	/* Get items from lifo */
+	for (i = 0U; i < LIST_LEN; i++) {
+		rx_data = k_lifo_get(&lifo, K_NO_WAIT);
+		zassert_equal(rx_data->value, lifo_data[LIST_LEN-1-i].value);
+	}
+
+	/* Clear the spawn thread */
+	k_thread_abort(tid);
+
+}
+
+/* STACK SCENARIO */
+/* Thread A enters items into a stack, starts thread B and waits for a semaphore. */
+/* Thread B extracts all items from the stack and enters some items back into the stack. */
+/* Thread B gives the semaphore for Thread A to continue. */
+/* Once the control is returned back to Thread A, it extracts all items from the stack. */
+/* Verify the data's correctness. */
+
+
+#define STACK_LEN	8
+#define MAX_ITEMS	8
+
+K_STACK_DEFINE(stack, STACK_LEN);
+stack_data_t stack_data[MAX_ITEMS];
+
+
+static void thread_entry_fn_stack(void *p1, void *p2, void *p3)
+{
+	stack_data_t *rx_data;
+	stack_data_t data[MAX_ITEMS];
+
+	/* fill data to compare */
+	for (int i = 0; i < STACK_LEN; i++) {
+		data[i] = i;
+	}
```

Comment:
What's the reason for using a different comparison approach compared to the first two cases?

Reply:
There's no reason. Just a proposal of doing something with data.